specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+-   Allow unpackaged apps to Update/Delete their local notifications.
+-   Allow packaged applications to process notification activators.
+-   Allow unpackaged apps to process notification activators.
+
+# Examples
+
+## Packaged App Registering and handling a notification
+
+This scenario is specifically geared towards Packaged apps that need to send an app notification. An app
+Developer Registers their app for notifications and sets it up for background\foreground
+activation. Adding a AppNotificationActivated event handler signals that the app is ready to accept and handle
+an action in the foreground where the user clicks on a Notification in the action centre or a UI element in
+the payload body (like a Button). The developer can inturn run some business logic in their app in
+response to the UI Action.
+
+```cpp
+
+CLSID const c_comServerId = { 0x43136EB5,0xD36C,0x11CF,{0xAD,0xBC,0x00,0xAA,0x00,0xA8,0x00,0x33} };
+
+int main()
+{
+    AppNotificationActivationInfo activationInfo{ winrt::guid(c_comServerId) };
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in Foreground. So intercept toast activators via Foreground event
+        const auto token = AppNotificationManager::Default().AppNotificationActivated([](const auto&, const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+        {
+            ProcessNotificationArgs(notificationActivatedEventArgs);
+        });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    }
+
+    return 0;
+}
+```
+
+## Unpackaged app Registering and handling a notification
+
+Similar to packaged apps, unpackaged apps can also Register to send app notifications and act as
+targets for activation events. There is one basic difference in the Registration for
+Unpackaged apps versus their packaged counterparts: they will need to specify their own friendly App
+DisplayName and IconUri since they don't have an app manifest file that contains the same.
+
+```cpp
+int main()
+{
+    PCWSTR const c_iconName = L"icon.png";
+    std::filesystem::path binaryPath = wil::GetModuleFileNameW<std::wstring>(nullptr);
+    std::filesystem::path directoryPath = binaryPath.parent_path();
+    std::filesystem::path iconPath = directoryPath / std::filesystem::path(c_iconName);
+
+    Uri iconUri(iconPath.wstring().c_str());
+    AppNotificationActivationInfo activationInfo(L"SampleApp", iconUri);
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
```

Comment:
I'm not sure the name `RegisterActivator` is the best name for this method, for unpackaged apps, isn't this also registering the app to send notifications, since it's registering the display name and icon URI (two things that have nothing to do with activation and purely have to do with sending/displaying notifications)?

Maybe simply `Register(appInfo)` would work? And then correspondingly, rename `AppNotificationActivationInfo` to `AppNotificationAppInfo` or `AppNotificationRegistrationInfo`, since it's more than just activation info?

Reply:
We will call it Register.

Reply:
DISCUSSION: Single Register() with no parameters.